crates/store/src/server/api.rs

```diff
@@ -519,6 +524,35 @@ impl api_server::Api for StoreApi {
 
         Ok(Response::new(GetAccountStateDeltaResponse { delta }))
     }
+
+    #[instrument(
+        target = COMPONENT,
+        name = "store.server.get_unconsumed_network_notes",
+        skip_all,
+        err
+    )]
+    async fn get_unconsumed_network_notes(
+        &self,
+        request: Request<generated::requests::GetUnconsumedNetworkNotesRequest>,
+    ) -> Result<Response<GetUnconsumedNetworkNotesResponse>, Status> {
+        let request = request.into_inner();
+        let state = self.state.clone();
+
+        let (notes, next_page) = state
+            .get_unconsumed_network_notes(Page::new(
+                request.page,
+                NonZero::try_from(request.limit as usize).map_err(|err: TryFromIntError| {
+                    invalid_argument(format!("Invalid limit: {err}"))
+                })?,
```

Comment:
nit: I'd maybe move this code above line 541 to make things a bit more readable.